src/bundle/ui-dev/src/modules/common/thumbnail/thumbnail.tsx

```diff
@@ -1,9 +1,15 @@
-import React from 'react';
-import PropTypes from 'prop-types';
-
 import Icon from '../icon/icon';
 
-const Thumbnail = ({ thumbnailData, iconExtraClasses, contentTypeIconPath }) => {
+interface ThumbnailProps {
+    thumbnailData: {
+        mimeType: string;
+        resource: string;
+    };
+    iconExtraClasses?: string;
+    contentTypeIconPath?: string | null;
+}
+
+const Thumbnail = ({ thumbnailData, iconExtraClasses = '', contentTypeIconPath = null }: ThumbnailProps) => {
```

Comment:
Should we consider adding return types or in this case do we prefer to rely on type inference?

Reply:
Do you know any online resources debating for or against this? :)

Reply:
Here are some links showing different perspectives among the community:
1. https://www.reddit.com/r/typescript/comments/g9olpy/when_should_you_write_a_type_and_when_should_you/?rdt=36868
2. https://stackoverflow.com/questions/45911407/type-inference-vs-explicit-type-declaration-in-typescript
3. https://www.devoreur2code.com/blog/type-inference-with-typescript